### PR TITLE
Pn/feature topic comment

### PR DESCRIPTION
**Pull request by nemoipaha**

### CODE REVIEW

File: resources/assets/js/app/views/comments/TopicCommentNew.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+                return App.getBaseUrl() + _.result(view.model, 'url') + _.result(attachModel, 'url');
+            },
+            method: 'post',
+            // input file name, registered on server
+            paramName: "f",
+            parallelUploads : 10,
+            hiddenInputContainer: '#drop',
+            autoProcessQueue : false,
+            maxFilesize: 8,
+            maxFiles: 5,
```
Comment (binaryvo) [inline resources/assets/js/app/views/comments/TopicCommentNew.js]: I think that parameters like:
- parallelUploads
- maxFilesize
- maxFiles 
- acceptedFiles
  should be moved to config file

File: resources/assets/js/app/views/comments/TopicCommentNew.js [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+            //if max files count
+            maxfilesexceeded: function (file) {
+                this.removeFile(file);
+                view.$('.errors').text('Max files is 5');
```
Comment (binaryvo) [inline resources/assets/js/app/views/comments/TopicCommentNew.js]: Move '.errors' to ui property please

File: resources/assets/js/app/views/comments/TopicCommentNew.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+            },
+            success: function (file, xhr) {
+                if (xhr.data) {
+                    view._files.push(xhr.data);
+                }
+            },
+            removedfile: function (file) {
+                if (file.id) {
+                    view.$(file.previewElement).remove();
+                    view.$('.dz-message').hide();
```
Comment (binaryvo) [inline resources/assets/js/app/views/comments/TopicCommentNew.js]: Please move to ui property

**Comment by binaryvo** [comment]
Please fix rest of remarks, update from next and make new PR

**Comment by nemoipaha** [comment]
all remarks fixed

File: app/Http/Controllers/TopicController.php [hunk 4]
```diff
@@ -22,6 +22,31 @@ class TopicController extends ApiController
      * @param Topic $topic
      * @return array
      */
+
+//    private function getItemMetaData($topic)
```
Comment (TatsuUkraine) [inline app/Http/Controllers/TopicController.php]: remove those commented methods if you don't need them anymore

**Comment by TatsuUkraine** [comment]
bug: when you remove last sub comment, button that shows sub comments still rendered

File: resources/assets/js/app/views/comments/TopicCommentNew.js [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+            paramName: "f",
+            hiddenInputContainer: '#drop',
+            autoProcessQueue : false,
+            parallelUploads : config.parallelFileUploads,
+            maxFilesize: config.maxFileSize,
+            maxFiles: config.maxFiles,
+            //if max files count
+            maxfilesexceeded: function (file) {
+                this.removeFile(file);
+                view.ui.errors.text('Max files is 5');
```
Comment (binaryvo) [inline resources/assets/js/app/views/comments/TopicCommentNew.js]: Please use value from config in error text: 
view.ui.errors.text('Max files is 5');
With data type specifying.

File: resources/assets/js/app/views/comments/TopicCommentNew.js [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+            },
+            uploadMultiple: false,
+            addRemoveLinks: true,
+            acceptedFiles: 'image/*,.pdf,.docx,.doc,.xlsx,.xls',
```
Comment (binaryvo) [inline resources/assets/js/app/views/comments/TopicCommentNew.js]: Please move acceptedFiles value to a config file:
acceptedFiles: 'image/*,.pdf,.docx,.doc,.xlsx,.xls',

**Comment by binaryvo** [comment]
bug: reply for comment isn't visible after adding. Only after page reload

**Comment by binaryvo** [comment]
Please fix issues, resolve all remarks and re-open PR

**Comment by nemoipaha** [comment]
all remarks are resolved

**Comment by binaryvo** [comment]
1. Reply for comments works fine. But I think that would be great to open 2nd level comments section after first commentary adding.
2. Please change title and submit button text in case of edit commentary: [picture](http://storage1.static.itmages.com/i/16/0913/h_1473742877_8429792_36d3e28aaf.png)
3. An error: [picture](http://pix.toile-libre.org/?img=1473742950.png)

**Comment by nemoipaha** [comment]
all errors fixed

**Comment by binaryvo** [comment]
Functionality works for me. But some remarks:
- Add title to modal window of create/edit comment: "Create comment" / "Edit comment" accordingly
- Add button 'Cancel' to to modal window of create/edit comment

**Comment by nemoipaha** [comment]
remarks fixed

**Pull request merged**